magento/magento2
# Fix admin localized date formatting

### Description (*)
There were 2 issues going on that are fixed by this PR. 

1. **Incorrect initial value parsing**
When parsing the initial value, say `2020-03-17`, the date was getting parse with the following format (for nl_NL): `dd-MM-y`. The value used was derived from the variable `outputDateFormat`. I changed the format by using the variable `inputeDateFormat`, which is `yyyy-MM-dd`.
2. **Cyclic event chain**
Variable `shiftedValue` was initialized by the `onValueChange` listener. But when changing `shiftedValue`, the `value` variable is changed by the `onShiftedValueChange` listener. This would mess up the date value as well. 
A cyclic event chain like the following could occur:
    - Change `shiftedValue` to `2020-03-17`.
    - Change `value` to `17-03-2020`.
    - Change `shiftedValue` to `20-03-2017`. 

### Fixed Issues (if relevant)
1. fixes magento/magento2#27310: Localized date field formatting in admin messing up actual date

### Manual testing scenarios (*)
1. Have 2 admin users. One on locale `en_US`, one on `nl_NL`.
2. Log in on `nl_NL` admin user.
3. Create a cart price rule, no coupon code needed, just a simple cart price rule. Make sure you set the **from date** (e.g. 17 march 2020). Save the cart price rule.
4. Check if the **from date** is correct.

### Contribution checklist (*)
